File: 20-valid-parentheses/20-valid-parentheses.c


bool isValid(char * s){
    char stack[10001];
    int top=-1;
    for (int i=0;i<strlen(s);i++){
        if (s[i]=='('||s[i]=='['||s[i]=='{')
            stack[++top]=s[i];
        else if(top==-1){
            return false;
        }
        else if(s[i]==')'){
            if (stack[top--]!='(')
                return false;
        }
        else if(s[i]=='}'){
            if (stack[top--]!='{')
                return false;
        }
        else if(s[i]==']'){
            if (stack[top--]!='[')
                return false;
        }
    }
    if (top==-1)
        return true;
    else 
        return false;
}

File: 20-valid-parentheses/20-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack=[]
        open=['(','{','[']
        for c in s:
            if c in open:
                stack.append(c)
                continue
            if len(stack)==0:
                return False
            if c==']':
                popped=stack.pop()
                if popped!='[':
                    return False
            elif c=='}':
                popped=stack.pop()
                if popped!='{':
                    return False
            elif c==')':
                popped=stack.pop()
                if popped!='(':
                    return False
            else:
                return False
        if len(stack)==0:
            return True
            



        